src/proxy.c

```diff
@@ -689,10 +683,55 @@ int parse_json_from_stdin (const char *section) {
 
 void print_usage()
 {
-    g_print("dbus-proxy, version %s\n", PACKAGE_VERSION);
-    g_print("Usage: dbus-proxy address session|system\n"
-            "waits for JSON conf at stdin.\n");
+    g_message("dbus-proxy, version %s\n", PACKAGE_VERSION);
+    g_message("Usage: dbus-proxy address session|system\n"
+              "waits for JSON conf at stdin.\n");
+}
+
+void handle_sigchld(int sig) {
+  while (waitpid((pid_t)(-1), 0, WNOHANG) > 0) {}
+}
+
+
+#ifdef LOG_TO_FILE
+FILE *log_file;
+
+gboolean log_file_is_open() {
+    return log_file ? TRUE : FALSE;
+}
+
+gboolean open_log_file() {
+    log_file = fopen("/tmp/dbus-proxy-log", "a");
+
+    if (log_file == NULL) {
+        return FALSE;
+    }
+
+    return TRUE;
+}
+
+gboolean close_log_file() {
```

Comment:
Perhaps it would make sense to check if the `log_file` is open before you close it?